File: src/main/java/yunas/Json.java
package yunas;

import com.google.gson.Gson;
import com.google.gson.JsonObject;

/**
 * Wrap Third Json Lib. (Gson).
 *
 * @author yosuke kobayashi
 */
public class Json<T> {

    public String toJson(Object obj) {
        return new Gson().toJson(obj);
    }

    public T fromJson(String jsonString , Class<T> clazz) {
        return new Gson().fromJson(jsonString, clazz);
    }

    public JsonObject fromJson(String jsonString){
        return new Gson().fromJson(jsonString, JsonObject.class);
    }
}

File: src/main/java/yunas/Json.kt
package yunas

import com.google.gson.Gson
import com.google.gson.JsonObject

/**
 * Wrap Third Json Lib. (Gson).
 *
 * @author yosuke kobayashi
 */
class Json<T> {

    fun toJson(obj: Any): String {
        return Gson().toJson(obj)
    }

    fun fromJson(jsonString: String, clazz: Class<T>): T {
        return Gson().fromJson(jsonString, clazz)
    }

    fun fromJson(jsonString: String): JsonObject {
        return Gson().fromJson(jsonString, JsonObject::class.java)
    }
}
